eng/targets/CSharp.Common.targets

```diff
@@ -36,7 +36,7 @@
 
     <!-- Enable .NET code style analysis during build for src projects. -->
     <!-- Workaround bug where turning this on produces warnings in VS -->
-    <EnforceCodeStyleInBuild Condition="'$(BuildingInsideVisualStudio)' == 'true' AND '$(VisualStudioVersion)' &lt; '17.0'">false</EnforceCodeStyleInBuild>
+    <EnforceCodeStyleInBuild Condition="'$(MSBuildRuntimeType)' != 'Core' AND $([MSBuild]::VersionLessThan('$(VisualStudioVersion)', '17.0'))">false</EnforceCodeStyleInBuild>
```

Comment:
And, was I correct we should enforce code style when building w/ desktop `msbuild`❔